mathics/core/parser/parser.py

```diff
@@ -308,13 +308,33 @@ def p_RawLeftAssociation(self, token):
     def p_LeftRowBox(self, token):
         self.consume()
         children = []
-        self.box_depth += 1
-        self.bracket_depth += 1
+        # If this does not happend, it would be because
+        # it was called when a `FormBox` was found.
+        if token.tag == "LeftRowBox":
+            self.box_depth += 1
+            self.bracket_depth += 1
```

Comment:
Every place there is some depth that increases by one, there should be another place that decreases the depth by one. 

Where is that?

Reply:
If the tat is not ```LeftRowBox```  then we never reach the decrement.

Reply:
You are saying something more or less like what the code says. Why is the decrement not reached. What is the depth used for?

Reply:
> You are saying something more or less like what the code says. Why is the decrement not reached. What is the depth used for?

The deep was used to handle nested expressions like ```\(  a b  \( InputForm \`   \(OutputForm \` c d\) e  \) \) ```

When a ``` \` ``` is found, then the previous tokens are collected, and evaluated as a `Symbol`, a  Box expression or `StandardForm` if there is no token before ``` \` ```. The result of this step is stored in a variable, to be used as the second argument of `FormBox`. 
 Then, a new call to the method is done, with `FormBox` as a tag.  In the new call, the next elements are processed until the bracket and box deeps reach 0. Then, these elements are collected into a `String` or a `RowBox`. Then, the instance where ``` \` ``` was found takes the format and the other collected elements, builds a ```FormBox[]```, and returns it (before reaching the line where the decrements happen). 


I tried to find a way to put the increment and the decrement together, but it just was more involved than this approach: I should have allowed the increment of the deep variables, and then doing the decrement before returning the FormBox. I didn't think that it was clearer.

Reply:
I hope the new comment helps to understand the logic.

Reply:
Thanks for the description. Something about this seems a little to complicated and that there should be something that follows the specification in a little more direct way: 

>  * In [InputForm](https://reference.wolfram.com/language/ref/InputForm.html) and [StandardForm](https://reference.wolfram.com/language/ref/StandardForm.html), \(form∖`input∖) yields [FormBox](https://reference.wolfram.com/language/ref/FormBox.html)[input,form].
> * ∖(∖`input∖) yields [FormBox](https://reference.wolfram.com/language/ref/FormBox.html)[input,RawForm].

I would like to think about and reflect on this some more. I think we can make this cleaner, clearer and simpler.

What would be ideal is if we could do a cleanup step first - no changes to behavior yet. 
Then add the list of tests that are wrong and that we want to fix. 

And after this, _then_ come up with a way to implement this.

Reply:
> > * ∖(∖`input∖) yields [FormBox](https://reference.wolfram.com/language/ref/FormBox.html)[input,RawForm].
> 

Actually, in WMA,
```
In[1]:= \(\`input\)                                                             

Out[1]= FormBox[input, StandardForm]
```



> I would like to think about and reflect on this some more. I think we can make this cleaner, clearer and simpler.
> 

I am sure it is possible, just that after trying for a while, I didn't find a better way. 

> What would be ideal is if we could do a cleanup step first - no changes to behavior yet. Then add the list of tests that are wrong and that we want to fix.
> 

What would be a cleanup step? The list of (simple) tests to check are the tests that I added in test/core/parser/test_parser.py. 


> And after this, _then_ come up with a way to implement this.

OK, in that case, I am going to put this as a draft for a while.